src/audio/base_fw.c

```diff
@@ -234,12 +236,21 @@ static int basefw_mem_state_info(uint32_t *data_offset, char *data)
 	return 0;
 }
 
+/* Assumed time interval between host sends timestamp and fw process it */
+#define TIME_SETTING_LATENCY_NS	120000
```

Comment:
Guessing is worse than doing nothing because it will sometimes ahead and sometimes behind, super confusing. Please leave this to zero so the skew is always one way.

Please correct me: in general I think it's pretty obvious that receiving some IPC happened _after_ it was sent, I mean neither IPC3 not IPC4 has a lot of concurrency, has any? 

(also: don't leave 30 lines between the definition and the only use)

Reply:
this is a common issue that host builds the time and sends to fw which process it a little later. At the time fw receives the timestamp, the timestamp is out of date since host time is going forward. The latency value is built from many experiments.  It is not accurate and universal. I have a better idea to do it in kernel. Let's wait for it.
````
host build timestamp ------>  host send timestamp by ipc msg -------> fw recieve and process it
             |   <-------          60 us        --------> |   <--------       about 60 us      ------>      |
````

Reply:
please check https://github.com/thesofproject/linux/pull/4706. Kernel will build the latency

Reply:
You're assuming the IPC latency is symmetric. It's clearly not, the Linux driver deals with a single interrupt and will check stream, IPC and SoundWire (in that order). If the host driver is busy with a SoundWire IPC, the IPC will be handled *after* renabling the global interrupt and going back to the interrupt thread.

On the firmware side, one would hope that Zephyr has a real-time behavior, but IIRC the IPCs are only processed at every firmware tick to avoid disrupting processing, so there some scheduling delay to be added too.

In other words, averaging the IPC latency is just as bad as adding a random fudge factor to the timestamps. The goal is not precise configuration but a means to correlate kernel and firmware. I don't think there's an expectation that there is a better than 1ms alignment, is there?

Back to requirements I guess :-)

Reply:
@plbossart  sure, I fully understand your assumption. But consider the real usage: when we sync timestamp ? Driver just do it after all audio initialization is done after powered on, At this time we don't have any IPC or any audio stream, and FW just boot up and have nothing to do.  Yes, it is still not very accurate, but without it we will found FW receive IPC msg before kernel send it. This will confuse developer.  Currently we have a time error of 100 us,  if 1ms error can expected, I am happy to remove the calibration.